### PR TITLE
Youtube links

### LINKED ISSUE #80: Add instructions/verify for youtube links (#54)
#54 caused by youtube links in wrong format. User must enter youtu.be link obtained by clicking share button - include instructions in form and link verification (aka show input error, build into MUI Input component, on bad entry)

Good:
- https://youtu.be/d0tRKmMC7po

Bad:
- https://www.youtube.com/watch?list=PLK386zwmYujQYy8C0wN5lNElf_rAd9uGv&v=nHkc1gTNdgI
- https://www.youtube.com/watch?v=oz6aSdruanw

### PR DESCRIPTION
**Pull request by LucasNVarella**
Fixes #80 and #54 .

Automatically converts 'bad' links from #80 to sharable links.

### CODE REVIEW

File: client/FileUpload.js [hunk 1]
```diff
@@ -54,6 +54,28 @@ class FileUpload extends Component {
...
+
+        this.handleLinkBLur = event => {
+            if (this.props.fileType === 'Video') {
+                let fileDoc = this.state.fileDoc;
+                let url = event.target.value;
+                if (!url.includes('youtu.be/')) { // Not a sharable link
+                    if (url.includes('youtube.com/watch?v='))  {
+                        // ...but we may be able to fix it
+                        let videoId;
+                        if (url.includes('&')) { // link includes extra params - remove them
```
Comment (matticoli) [inline client/FileUpload.js:66]: this doesn't account for if `v=` is not first param

File: client/FileUpload.js [hunk 2]
```diff
@@ -142,8 +164,10 @@ class FileUpload extends Component {
                     style={{margin: 5}}
                     value={(this.state.fileDoc && this.state.fileDoc[url]) || ""}
                     onChange={this.handleTextChange}
+                    onBlur={this.handleLinkBLur}
```
Comment (matticoli) [inline client/FileUpload.js:167]: Why onBlue instead of onChange? also typo in function name

Comment (LucasNVarella) [reply client/FileUpload.js:167]: onChange fires every time you type into the box, so if for some reason a user were to manually type in a link, onBlur makes it so the box waits till they're done before converting the link.
Fixed typo

File: client/FileUpload.js (hunk 1 above)
Comment (LucasNVarella) [reply client/FileUpload.js:66]: You're right, I completely missed that. Fixed

File: client/FileUpload.js [hunk 3]
```diff
@@ -54,6 +54,28 @@ class FileUpload extends Component {
             }
             this.setState({fileDoc: fileDoc});
         };
+
+        this.handleLinkBLur = event => {
+            if (this.props.fileType === 'Video') {
+                let fileDoc = this.state.fileDoc;
+                let url = event.target.value;
+                if (!url.includes('youtu.be/')) { // Not a sharable link
```
Comment (matticoli) [inline client/FileUpload.js:62]: `.startsWith(...)`

File: client/FileUpload.js (hunk 1 above)
Comment (matticoli) [reply client/FileUpload.js]: You can use regex to get video id, so you don't have to worry about covering all the edge cases. There can be any number of get params in video (maybe user has browser ext, etc)
```
let regex = /\?.*v=([^&]*)(\&|$)/gi;
let match = regex.exec(url);
let id=match[0];
```

File: client/FileUpload.js (hunk 2 above)
Comment (matticoli) [reply client/FileUpload.js]: func name still has a capital L but it's whatever

File: client/FileUpload.js (hunk 1 above)
Comment (LucasNVarella) [reply client/FileUpload.js]: Had to fix a bug in the regex, but was able to swap my previous code for it. Thanks!

File: client/FileUpload.js (hunk 2 above)
Comment (LucasNVarella) [reply client/FileUpload.js]: It doesn't, the preview on this comment hasn't updated

File: client/FileUpload.js (hunk 3 above)
Comment (LucasNVarella) [reply client/FileUpload.js:62]: to use .startsWith() we'd have to account for multiple cases: youtu.be, https://youtu.be, and possibly more depending on where the user got the link. Probably best to stick with includes

**Pull request merged**